diff -rupN giza-pp/GIZA++-v2/file_spec.h giza-pp.new/GIZA++-v2/file_spec.h
--- giza-pp/GIZA++-v2/file_spec.h	2009-03-20 07:41:12.000000000 -0400
+++ giza-pp.new/GIZA++-v2/file_spec.h	2010-07-18 22:16:40.000000000 -0400
@@ -37,13 +37,13 @@ char *Get_File_Spec (){
   struct tm *local;
   time_t t;
   char *user;
-  char time_stmp[17];
+  char time_stmp[19];
   char *file_spec = 0;
   
   t = time(NULL);
   local = localtime(&t);
   
-  sprintf(time_stmp, "%02d-%02d-%02d.%02d%02d%02d.", local->tm_year, 
+  sprintf(time_stmp, "%04d-%02d-%02d.%02d%02d%02d.", 1900 + local->tm_year, 
 	  (local->tm_mon + 1), local->tm_mday, local->tm_hour, 
 	  local->tm_min, local->tm_sec);
   user = getenv("USER");
Binary files giza-pp/GIZA++-v2/GIZA++ and giza-pp.new/GIZA++-v2/GIZA++ differ
diff -rupN giza-pp/GIZA++-v2/Makefile giza-pp.new/GIZA++-v2/Makefile
--- giza-pp/GIZA++-v2/Makefile	2009-03-20 08:01:31.000000000 -0400
+++ giza-pp.new/GIZA++-v2/Makefile	2010-07-18 22:49:35.000000000 -0400
@@ -5,7 +5,7 @@ INSTALLDIR ?= /usr/local/bin/
 CXX = g++
 
 CFLAGS = $(CFLAGS_GLOBAL) -Wall -W -Wno-deprecated
-CFLAGS_OPT = $(CFLAGS) -O3 -DNDEBUG -DWORDINDEX_WITH_4_BYTE -DBINARY_SEARCH_FOR_TTABLE
+CFLAGS_OPT = $(CFLAGS) -O3 -DNDEBUG -DWORDINDEX_WITH_4_BYTE
 CFLAGS_PRF = $(CFLAGS) -O2 -pg -DNDEBUG -DWORDINDEX_WITH_4_BYTE
 CFLAGS_DBG = $(CFLAGS) -g -DDEBUG -DWORDINDEX_WITH_4_BYTE
 CFLAGS_NRM = $(CFLAGS) -DWORDINDEX_WITH_4_BYTE
Binary files giza-pp/GIZA++-v2/optimized/alignment.o and giza-pp.new/GIZA++-v2/optimized/alignment.o differ
Binary files giza-pp/GIZA++-v2/optimized/AlignTables.o and giza-pp.new/GIZA++-v2/optimized/AlignTables.o differ
Binary files giza-pp/GIZA++-v2/optimized/ATables.o and giza-pp.new/GIZA++-v2/optimized/ATables.o differ
Binary files giza-pp/GIZA++-v2/optimized/collCounts.o and giza-pp.new/GIZA++-v2/optimized/collCounts.o differ
Binary files giza-pp/GIZA++-v2/optimized/Dictionary.o and giza-pp.new/GIZA++-v2/optimized/Dictionary.o differ
Binary files giza-pp/GIZA++-v2/optimized/ForwardBackward.o and giza-pp.new/GIZA++-v2/optimized/ForwardBackward.o differ
Binary files giza-pp/GIZA++-v2/optimized/getSentence.o and giza-pp.new/GIZA++-v2/optimized/getSentence.o differ
Binary files giza-pp/GIZA++-v2/optimized/hmm.o and giza-pp.new/GIZA++-v2/optimized/hmm.o differ
Binary files giza-pp/GIZA++-v2/optimized/HMMTables.o and giza-pp.new/GIZA++-v2/optimized/HMMTables.o differ
Binary files giza-pp/GIZA++-v2/optimized/main.o and giza-pp.new/GIZA++-v2/optimized/main.o differ
Binary files giza-pp/GIZA++-v2/optimized/model1.o and giza-pp.new/GIZA++-v2/optimized/model1.o differ
Binary files giza-pp/GIZA++-v2/optimized/model2.o and giza-pp.new/GIZA++-v2/optimized/model2.o differ
Binary files giza-pp/GIZA++-v2/optimized/model2to3.o and giza-pp.new/GIZA++-v2/optimized/model2to3.o differ
Binary files giza-pp/GIZA++-v2/optimized/model345-peg.o and giza-pp.new/GIZA++-v2/optimized/model345-peg.o differ
Binary files giza-pp/GIZA++-v2/optimized/model3.o and giza-pp.new/GIZA++-v2/optimized/model3.o differ
Binary files giza-pp/GIZA++-v2/optimized/model3_viterbi.o and giza-pp.new/GIZA++-v2/optimized/model3_viterbi.o differ
Binary files giza-pp/GIZA++-v2/optimized/model3_viterbi_with_tricks.o and giza-pp.new/GIZA++-v2/optimized/model3_viterbi_with_tricks.o differ
Binary files giza-pp/GIZA++-v2/optimized/MoveSwapMatrix.o and giza-pp.new/GIZA++-v2/optimized/MoveSwapMatrix.o differ
Binary files giza-pp/GIZA++-v2/optimized/myassert.o and giza-pp.new/GIZA++-v2/optimized/myassert.o differ
Binary files giza-pp/GIZA++-v2/optimized/NTables.o and giza-pp.new/GIZA++-v2/optimized/NTables.o differ
Binary files giza-pp/GIZA++-v2/optimized/Parameter.o and giza-pp.new/GIZA++-v2/optimized/Parameter.o differ
Binary files giza-pp/GIZA++-v2/optimized/parse.o and giza-pp.new/GIZA++-v2/optimized/parse.o differ
Binary files giza-pp/GIZA++-v2/optimized/Perplexity.o and giza-pp.new/GIZA++-v2/optimized/Perplexity.o differ
Binary files giza-pp/GIZA++-v2/optimized/reports.o and giza-pp.new/GIZA++-v2/optimized/reports.o differ
Binary files giza-pp/GIZA++-v2/optimized/transpair_model3.o and giza-pp.new/GIZA++-v2/optimized/transpair_model3.o differ
Binary files giza-pp/GIZA++-v2/optimized/transpair_model4.o and giza-pp.new/GIZA++-v2/optimized/transpair_model4.o differ
Binary files giza-pp/GIZA++-v2/optimized/transpair_model5.o and giza-pp.new/GIZA++-v2/optimized/transpair_model5.o differ
Binary files giza-pp/GIZA++-v2/optimized/TTables.o and giza-pp.new/GIZA++-v2/optimized/TTables.o differ
Binary files giza-pp/GIZA++-v2/optimized/utility.o and giza-pp.new/GIZA++-v2/optimized/utility.o differ
Binary files giza-pp/GIZA++-v2/optimized/vocab.o and giza-pp.new/GIZA++-v2/optimized/vocab.o differ
Binary files giza-pp/GIZA++-v2/plain2snt.out and giza-pp.new/GIZA++-v2/plain2snt.out differ
Binary files giza-pp/GIZA++-v2/snt2cooc.out and giza-pp.new/GIZA++-v2/snt2cooc.out differ
Binary files giza-pp/GIZA++-v2/snt2plain.out and giza-pp.new/GIZA++-v2/snt2plain.out differ
Binary files giza-pp/mkcls-v2/GDAOptimization.o and giza-pp.new/mkcls-v2/GDAOptimization.o differ
Binary files giza-pp/mkcls-v2/general.o and giza-pp.new/mkcls-v2/general.o differ
Binary files giza-pp/mkcls-v2/HCOptimization.o and giza-pp.new/mkcls-v2/HCOptimization.o differ
Binary files giza-pp/mkcls-v2/IterOptimization.o and giza-pp.new/mkcls-v2/IterOptimization.o differ
Binary files giza-pp/mkcls-v2/KategProblemKBC.o and giza-pp.new/mkcls-v2/KategProblemKBC.o differ
Binary files giza-pp/mkcls-v2/KategProblem.o and giza-pp.new/mkcls-v2/KategProblem.o differ
Binary files giza-pp/mkcls-v2/KategProblemTest.o and giza-pp.new/mkcls-v2/KategProblemTest.o differ
Binary files giza-pp/mkcls-v2/KategProblemWBC.o and giza-pp.new/mkcls-v2/KategProblemWBC.o differ
Binary files giza-pp/mkcls-v2/mkcls and giza-pp.new/mkcls-v2/mkcls differ
Binary files giza-pp/mkcls-v2/mkcls.o and giza-pp.new/mkcls-v2/mkcls.o differ
Binary files giza-pp/mkcls-v2/MYOptimization.o and giza-pp.new/mkcls-v2/MYOptimization.o differ
Binary files giza-pp/mkcls-v2/Optimization.o and giza-pp.new/mkcls-v2/Optimization.o differ
Binary files giza-pp/mkcls-v2/Problem.o and giza-pp.new/mkcls-v2/Problem.o differ
Binary files giza-pp/mkcls-v2/ProblemTest.o and giza-pp.new/mkcls-v2/ProblemTest.o differ
Binary files giza-pp/mkcls-v2/RRTOptimization.o and giza-pp.new/mkcls-v2/RRTOptimization.o differ
Binary files giza-pp/mkcls-v2/SAOptimization.o and giza-pp.new/mkcls-v2/SAOptimization.o differ
Binary files giza-pp/mkcls-v2/StatVar.o and giza-pp.new/mkcls-v2/StatVar.o differ
Binary files giza-pp/mkcls-v2/TAOptimization.o and giza-pp.new/mkcls-v2/TAOptimization.o differ
